### ISSUE TITLE
Element with id .... is already registered, tests not runnable

### ISSUE THREAD

**Issue by justfalter** [hbenl/vscode-test-explorer#24]
Experienced with: Test Explorer UI 2.2.0, Jasmine Test Explorer 1.3.0

I've found that under certain circumstances, the Test Explorer UI will fail to render unit tests elements due to an apparent id conflict. Specifically, expanding a file in the Test Explorer UI under the following circumstances results in an error `Element with id .... is already registered`, and the tests are neither navigable nor runnable:

### A spec file contains two identically-named describe blocks.
```
describe('Thing1', () => {
    it('should foo', pending);
});
describe('Thing1', () => {
    it('should bar', pending);
});
```
### Two spec files each contain identically-named describe blocks.
foo/bar/bla/file1.spec.js:
```
describe('Thing1', () => {
    it('should foo', pending);
});
```
oh/boy/file2.spec.js:
```
describe('Thing1', () => {
    it('should bar', pending);
});
```

### A describe block contains two identically-named tests
```
describe('Test1', () => {
    it('should foo', pending);
    it('should foo', pending);
});
```

**Comment by iMarv** [comment]
Having the same issue with jest, running tests with similar names aswell

**Comment by igoventura** [comment]
I'm using the version 2.12.0 and still getting this error

**Comment by hbenl** [comment]
@igoventura Please tell me the id contained in the error message

**Comment by igoventura** [comment]
@hbenl the message is `Element with id error is already registered`

**Comment by hbenl** [comment]
@igoventura Thanks, it's fixed in version 2.12.1.
